### ISSUE TITLE
Whitebox attack with ImageNet ignores normalization

### ISSUE THREAD

**Issue by prateekmalhotra** [Harry24k/adversairal-attacks-pytorch#1]
Hello,

I noticed that the Panda image is not normalized before feeding it to the Inception v3 network. Here is a link describing the required normalization: https://discuss.pytorch.org/t/how-to-preprocess-input-for-pre-trained-networks/683

Let me know what you think.

Prateek

**Comment by Harry24k** [comment]
Hi @prateekmalhotra,

You're right. I added the normalization from [https://discuss.pytorch.org/t/how-to-preprocess-input-for-pre-trained-networks/683](https://discuss.pytorch.org/t/how-to-preprocess-input-for-pre-trained-networks/683). The results are shown in [https://github.com/Harry24k/adversairal-attacks-pytorch/blob/master/demos/White%20Box%20Attack%20with%20Imagenet.ipynb](https://github.com/Harry24k/adversairal-attacks-pytorch/blob/master/demos/White%20Box%20Attack%20with%20Imagenet.ipynb). After adding normalization, only CW and PGD succeed to attack.

Thanks.

Harry.

**Comment by prateekmalhotra** [comment]
Thank you for your quick response! There's one thing that still might be missing: given that this library only works in creating adversarial examples for images **normalized between 0 and 1**, I don't think it's wise to generate examples after PyTorch's normalization. What do you think?

**Comment by Harry24k** [comment]
After posting the answers above, I was looking for the answer. To my best knowledge, there is no paper that gives us clear suggestions of this situation. I saw one simple technique that applying inverse-normalization in other codes. For example, if we subtract 1 and divide 2, on the contrary, we insert the process of multiplying 2 and adding 1.

However, in this case, I think it's more complicated when we use attack packages like torchattacks. Thus, I suggest other technique to add normalization layer to the model with nn.Sequential. I uploaded modified version of the demo [https://github.com/Harry24k/adversairal-attacks-pytorch/blob/master/demos/White%20Box%20Attack%20with%20Imagenet.ipynb](https://github.com/Harry24k/adversairal-attacks-pytorch/blob/master/demos/White%20Box%20Attack%20with%20Imagenet.ipynb). In this case, all attacks show successful results except FGSM nad RFSM. Thank you for the meaningful question. :).
